src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -138,14 +144,17 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
             with _ -> defaultMscorlibVersion
 
         // Get the ProductVersion of this framework compare with table yield compatible monikers
-        let _, _, _, _, moniker =
-            desktopProductVersionMonikers
-            |> Array.find (fun (major, minor, build, revision, _) ->
-                (majorPart >= major) &&
-                (minorPart >= minor) &&
-                (buildPart >= build) &&
-                (privatePart >= revision))
-        moniker
+        match desktopProductVersionMonikers
+              |> Array.tryFind (fun (major, minor, build, revision, _) ->
+                    (majorPart >= major) &&
+                    (minorPart >= minor) &&
+                    (buildPart >= build) &&
+                    (privatePart >= revision)) with
+        | Some (_,_,_,_,moniker) ->
+            moniker
+        | None ->
+            // no TFM could be found, assume latest stable?
+            "net48"
```

Comment:
@KevinRansom I think with the above numbers this shouldn't fail very often in practice, but I think we should have some sort of fallback here to prevent the compiler from just crashing.  Is this a sane default?